app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,8 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: [:show, :edit, :update, :destroy]
 
   # GET /posts/1
-  # GET /posts/1.json
   def show
+    @post ||= Post.find(params[:id])
```

Comment:
👍  but instead od repeating it in every action you can just create a private method

```
def post
   @post ||= Post.find(params[:id])
end
```

and use it in show, edit and update :)

Reply:
Ok ,this makes sense.

Whats the difference between this and our previous approach of calling `post` method in the `before_action` helper of the controller? Is this a Best Practice approach?

Reply:
I'd say that they are just diffrent approaches. None is really better than other... or some devs would prefer one, others another ;)

I like to mention it in controllers reviews because it is a perfect scenario to learn how useful  `||=`  can be.